canonical-web-and-design/maas-ui
# feat(ui): build ModelNotFound component

## Done

- Created a component `ModelNotFound` that standardises how we handle displaying the details page of a model that does not exists.
- Added it to device details

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Using bolla, navigate to the following links and check that the header and return link are correct:
  - [Machine not found](http://0.0.0.0:8400/MAAS/r/machine/abc123)
  - [Device not found](http://0.0.0.0:8400/MAAS/r/device/abc123)
  - [LXD host not found](http://0.0.0.0:8400/MAAS/r/kvm/lxd/12345)
  - [Virsh host not found](http://0.0.0.0:8400/MAAS/r/kvm/virsh/12345)
  - [LXD cluster not found](http://0.0.0.0:8400/MAAS/r/kvm/lxd/cluster/12345)
  - [LXD host (in cluster) not found](http://0.0.0.0:8400/MAAS/r/kvm/lxd/cluster/29/vms/12345)

## Fixes

Fixes canonical-web-and-design/app-tribe#534

## Screenshot

![Screenshot 2021-11-24 at 11-40-14 MAAS](https://user-images.githubusercontent.com/25733845/143155630-ff3a6184-525d-4183-b77b-65cf4ff51e11.png)
